nova/tests/unit/scheduler/filters/test_external_scheduler_filter.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 SAP SE or an SAP affiliate company.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+# Tests for external scheduler filter.
+
+from unittest.mock import MagicMock
+from unittest.mock import patch
+from unittest.mock import sentinel
+
+import nova.conf
+from nova import objects
+from nova.scheduler.filters import external_scheduler_filter
+from nova import test
+from nova.tests.unit.scheduler import fakes
+
+
+CONF = nova.conf.CONF
+
+
+class ExternalSchedulerFilterTestCase(test.NoDBTestCase):
```

Comment:
This is missing at least 3 tests:
a) we're disabled
b) we're enabled, but the api is unreachable (mock_post.side_effect = requests.RequestException() or something)
c) we're enabled, the api returns an empty reply
d) we're enabled, the api returns some error code (500? 503?) (is this this same as b? we could make sure we catch a broad-enough exception from requests)